cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterCreator.java

```diff
@@ -62,20 +62,22 @@ public class KafkaClusterCreator {
     /**
      * Constructor
      *
-     * @param vertx             Vert.x instance
-     * @param reconciliation    Reconciliation marker
-     * @param config            Cluster Operator configuration
-     * @param supplier          Resource Operators supplier
+     * @param vertx                     Vert.x instance
+     * @param reconciliation            Reconciliation marker
+     * @param config                    Cluster Operator configuration
+     * @param supplier                  Resource Operators supplier
+     * @param kafkaMetadataConfigState  Metadata state related to nodes configuration
      */
     public KafkaClusterCreator(
             Vertx vertx,
             Reconciliation reconciliation,
             ClusterOperatorConfig config,
-            ResourceOperatorSupplier supplier
+            ResourceOperatorSupplier supplier,
+            KafkaMetadataConfigurationState kafkaMetadataConfigState
```

Comment:
Would it make sense to move it before the supplier to group it with the configs rather than the operations?